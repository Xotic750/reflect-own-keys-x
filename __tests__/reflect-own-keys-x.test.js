let reflectOwnKeys;

if (typeof module === 'object' && module.exports) {
  require('es5-shim');
  require('es5-shim/es5-sham');

  if (typeof JSON === 'undefined') {
    JSON = {};
  }

  require('json3').runInContext(null, JSON);
  require('es6-shim');
  const es7 = require('es7-shim');
  Object.keys(es7).forEach(function(key) {
    const obj = es7[key];

    if (typeof obj.shim === 'function') {
      obj.shim();
    }
  });
  reflectOwnKeys = require('../../index.js');
} else {
  reflectOwnKeys = returnExports;
}

const hasSymbols = typeof Symbol === 'function' && typeof Symbol('') === 'symbol';
const ifSymbolsIt = hasSymbols ? it : xit;

describe('reflectOwnKeys', function() {
  it('is a function', function() {
    expect(typeof reflectOwnKeys).toBe('function');
  });

  it('throws if the target isnâ€™t an object', function() {
    expect(function() {
      reflectOwnKeys();
    }).toThrow();

    expect(function() {
      reflectOwnKeys(void 0);
    }).toThrow();

    expect(function() {
      reflectOwnKeys(null);
    }).toThrow();

    expect(function() {
      reflectOwnKeys(1);
    }).toThrow();

    expect(function() {
      reflectOwnKeys(true);
    }).toThrow();

    expect(function() {
      reflectOwnKeys('');
    }).toThrow();
  });

  it('should return the same result as Object.getOwnPropertyNames if there are no Symbols', function() {
    const obj = {bar: 1, foo: 2};
    obj[1] = 'first';
    const result = Object.getOwnPropertyNames(obj);
    // Reflect.ownKeys depends on the implementation of
    // Object.getOwnPropertyNames, at least for non-symbol keys.
    expect(reflectOwnKeys(obj)).toStrictEqual(result);
    // We can only be sure of which keys should exist.
    expect(result.sort()).toStrictEqual(['1', 'bar', 'foo']);
  });

  ifSymbolsIt('symbols come last', function() {
    const s = Symbol('');
    const o = {'non-symbol': true};
    o[s] = true;
    expect(reflectOwnKeys(o)).toStrictEqual(['non-symbol', s]);
  });
});
