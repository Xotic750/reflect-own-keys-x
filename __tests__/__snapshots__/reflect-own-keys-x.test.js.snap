// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reflectOwnKeys throws if the target isn’t an object 1`] = `"Reflect.ownKeys called on non-object"`;

exports[`reflectOwnKeys throws if the target isn’t an object 2`] = `"Reflect.ownKeys called on non-object"`;

exports[`reflectOwnKeys throws if the target isn’t an object 3`] = `"Reflect.ownKeys called on non-object"`;

exports[`reflectOwnKeys throws if the target isn’t an object 4`] = `"Reflect.ownKeys called on non-object"`;

exports[`reflectOwnKeys throws if the target isn’t an object 5`] = `"Reflect.ownKeys called on non-object"`;

exports[`reflectOwnKeys throws if the target isn’t an object 6`] = `"Reflect.ownKeys called on non-object"`;
